google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2402,7 +2394,7 @@ private void removeIdleSessions(Instant currTime) {
         // below MinSessions.
         Instant minLastUseTime = currTime.minus(options.getRemoveInactiveSessionAfter());
         Iterator<PooledSession> iterator = sessions.descendingIterator();
-        while (iterator.hasNext()) {
+        while (iterator.hasNext() && allSessions.size() > options.getMinSessions()) {
```

Comment:
Don't let the idle session remover remove so many sessions that we drop below `MinSessions`.